### DISCUSSION TITLE
PyAutoFEP is it possible to calculate the membrane system at present?

### DISCUSSION THREAD

**Discussion by Teng-Dan** [luancarvalhomartins/PyAutoFEP#28] [Q&A]
Hi, it is mentioned in the manual ‘It automates the building of a simple receptor-ligand system and will fail for complex cases (eg: membrane systems, systems with vacuum regions, solvent mixtures, and systems with nucleic acids on cofactors).’ So I want to know whether PyAutoFEP can be applied to the calculation of membrane system. Looking forward to your reply, thanks!

**Comment by luancarvalhomartins** [comment]
Yes, PyAutoFEP is able to handle membrane systems. The automated system builder, however, cannot. Therefore, you will have to prepare the system and supply it as a pre-solvated structure, ideally, alongside the topology. You will also have to use custom mdp files matching your membrane (semi-isotropic pressure coupling? anisotropic? compressibility?). I am planning to add a tutorial for membrane proteins anytime soon. In case you need further help, don't hesitate to contact me here or via email.

**Reply by Teng-Dan** [reply]
That's great! In fact, I don't know how to manually process and commit these files when using PyAutoFEP. The protein currently studied in my project is G-protein coupled receptor (GPCR), a very representative membrane protein. Can you consider GPCR as an example when preparing the membrane protein tutorial? Could you please let me know when you update the membrane protein tutorial, I am looking forward to your tutorial, thank you very much!

**Comment by luancarvalhomartins** [comment]
My plan is to use B2AR in the tutorial. I am a bit swamped right now, but I will do my best to work in this tutorial soon. In case you are interested, I can send you an alpha version of the aforementioned tutorial before commiting a polished version to the repo (feel free to reach me by email if so).

In case you are interested in moving forward regardless of the tutorial, I suggest you to (hopefully, it may also be useful to other users):
 - Process the PDB of the membrane protein (missing residues/atoms, cosolvents/crystallization species, protonation)
 - Use the OPM server (https://opm.phar.umich.edu/) to get the orientation of the protein
 - Prepare a bilipid+water+ions box (eg, a truncated hexagonal box)
 - Use `gmx mdrun -membed` to insert the protein in the membrane
 - Minimize, then equilibrate the system for, say, 50-100 ns
 - Use the last frame from this equilibration, along with its topology, as input to `prepare_dual_topology.py` using `pre_solvated`. Notes:
   - Currently, `prepare_dual_topology.py` won't run `gmx genion` for pre_solvated systems, so be aware of charge differences
   - Make sure to include any additional topology files as `extrafiles` or `extradirs`
   - In case you equilibrate the system with a ligand, currently, `prepare_dual_topology.py` requires that the residue name of this ligand to be `LIG`. I hope to solve this, but it's not high priority.

**Reply by Teng-Dan** [reply]
That's sound great! Looking forward to your tutorial. I'll try to run as you suggested first, thanks for your reply.
